### PR TITLE
Bug 1202003 - Make filter panel into a regular menu

### PR DESCRIPTION
**Pull request by camd**
Now that field filters are handled on the Active Filters bar,
we can make the filter panel into a menu.

To be honest, this makes my life easier when moving to Bootstrap 4 so I don't have to clean up as much stuff.  :)
screenshot 2017-10-06 16 35 29

### CODE REVIEW

File: ui/partials/main/thGlobalTopNavPanel.html [hunk 1]
```diff
@@ -80,17 +80,60 @@
...
+          <ul id="filter-dropdown"
+              class="dropdown-menu checkbox-dropdown-menu"
+              role="menu" aria-labelledby="filterLabel">
+            <span class="filtergroup" ng-repeat="(group_order, group) in filterGroups">
+              <li role="presentation" class="dropdown-header">
+                  {{::group.name}}
+                  <span class="fa fa-info-circle"></span>
+              </li>
+              <span ng-repeat="filterName in group.resultStatuses" >
+                <label class="checkbox-inline {{::checkClass}} dropdown-item">
```
Comment (edmorley) [inline ui/partials/main/thGlobalTopNavPanel.html]: The `checkbox-inline` class listed here no longer has any references in CSS, so can be removed if not needed :-)

File: ui/partials/main/thFilterPanel.html [hunk 2]
```diff
@@ -1,56 +0,0 @@
...
-            ng-click="thJobFilters.resetNonFieldFilters()">reset</span>
-    </span>
-  </div>
-
-  <!-- Result status filters -->
-  <span ng-repeat="group in filterGroups"
-        class="panel panel-default th-option-group th-inline-option-group">
-
-    <!-- Group heading -->
-    <div class="panel-heading th-option-heading">{{::group.name}}
```
Comment (edmorley) [inline ui/partials/main/thFilterPanel.html]: The `.th-top-nav-options-panel .th-option-heading` CSS rule in `treeherder-navbar-panels.css` seems unused after this

File: ui/partials/main/thFilterPanel.html [hunk 3]
```diff
@@ -1,56 +0,0 @@
-<div id="filter"
-     class="th-top-nav-options-panel model-body"
-     ng-controller="FilterPanelCtrl">
-  <div class="nav-panel-help-text">Specify which jobs to show based on this filter criteria.
```
Comment (edmorley) [inline ui/partials/main/thFilterPanel.html]: `.nav-panel-help-text` in `treeherder-narbar.css` seems unused after this

File: ui/partials/main/thGlobalTopNavPanel.html [hunk 4]
```diff
@@ -80,17 +80,60 @@
         </span>
       </span>
 
-      <!-- Global Filter Button -->
-      <span class="btn btn-view-nav btn-right-navbar nav-menu-btn"
-            title="Global job filters"
-            ng-class="{'active': (isFilterPanelShowing)}"
-            ng-click="setFilterPanelShowing(!isFilterPanelShowing)"
```
Comment (edmorley) [inline ui/partials/main/thGlobalTopNavPanel.html]: `setFilterPanelShowing()` and `isFilterPanelShowing` in `main.js` are unused after this

File: ui/js/controllers/filters.js [hunk 5]
```diff
@@ -37,22 +37,6 @@ treeherderApp.controller('FilterPanelCtrl', [
         ]);
 
         /**
-         * Handle checking the "all" button for a result status group
-         */
-        $scope.toggleResultStatusGroup = function (group) {
-            var check = function (rs) {
-                $scope.resultStatusFilters[rs] = group.allChecked;
```
Comment (edmorley) [inline ui/js/controllers/filters.js]: I think `group.allChecked` in `filters.js` in now used after this?

**Comment by camd** [comment]
Thanks for finding all that dead code.  Sorry I missed it... AGAIN!

**Pull request merged**